lib/record.js

```diff
@@ -85,16 +87,29 @@ exports.setSignedMode = function setSignedMode () {
  *
  * Sets the user to anonymous mode
  */
-exports.setAnonymousMode = function setAnonymousMode () {
+exports.setAnonymousMode = function setAnonymousMode (keepIdentifier) {
   if (this.client.storeConsentByLocalStorage) {
     global.localStorage.setItem(SIGNEDMODECOOKIE, 'false')
   } else {
     setCookie(this.client.storage, SIGNEDMODECOOKIE, 'false')
   }
 
-  this.resetUUID(this.client.storage, this.client.track.uuid)
-  // remove global cookie too
-  cookie.removeItem(this.client.globalIdCookie)
+  if (!keepIdentifier) {
```

Comment:
Why does this not follow the decentralized way currently employed by the SDK. For example: `setCookie(this.client.storage, NONSIGNEDMODEKEEPIDENTIFER, 'true')`. I'm not sure I understand the need to centralized cookie logic for cookie here in record.js

IMO, unless I'm missing some important info here, each logic should be moved back to its original plugin, with extra check of NONSIGNEDMODEKEEPIDENTIFER

- [ ] Review this logic

Reply:
With new cookie, it requires refreshing the page to apply the logic which might not be what we want.